ziafazal/SOL-1560:load db_fixures when run with --testsonly

ziafazal opened the pull request:
@benpatterson this PR enables loading of db fixues when bok-choy tests are run with `--testsonly`. This would fix this error. 
```
 FixtureError: Could not create course {'org': 'test_org', 'run': 'test_run', 'number': '80423875555327550007069674098531250060', 'display_name': 'Test Course13324843804671132797190073900964414758'}.  Status was 400
    Other stuff\: {
     "error": "You must link this course to an organization in order to continue. Organization you selected does not exist in the system, you will need to add it to the system"
   }
```
I have replaced the deprecated reset db command with flush too.

benpatterson commented:
@ziafazal Sorry if this is obvious, but did you reproduce the bug locally, and then confirm that this change fixes it?

benpatterson commented:
Ah I see that a11y has already failed. My guess is that this is because loaddata is being run twice now. You may need to remove it from [here](https://github.com/edx/edx-platform/pull/11097/files#diff-607194527ece372e33ca69e0371c3084R39) as part of this changeset.

ziafazal commented:
jenkins run a11y

ziafazal commented:
@benpatterson Yeah i have tested these changes locally. 
1) I deleted edxtest DB.
2) Removed accpetance/test/db_cache
3) run servers
` paver test_bokchoy --serversonly`
4) run test 3 times with this command after removing flaky decorator
`paver test_bokchoy -t discussion/test_discussion.py:DiscussionCommentEditTest.test_edit_comment_as_moderator --fasttest --testsonly
`

benpatterson commented:
@ziafazal ok, great. Glad you can reproduce the fix locally. 

However, I think this fix breaks the end-to-end scenario (that is, not-using `--testsonly` and `--serversonly`), as you can see from the broken builds.

ziafazal commented:
@benpatterson putting `reset_db` back in `scripts/reset-test-db.sh` makes this build green. It looks to me Jenkins passes PR when `scripts/reset-test-db.sh` is broken.

benpatterson commented:
There are two use cases:
1. Using the serversonly/testsonly combination
2. Running as a single process

You need to be sure both of those work on this pr. You should look at when
loaddata is being run...it should run one time in both cases.

sent via Android
On Dec 31, 2015 6:40 AM, "Zia Fazal" <notifications@github.com> wrote:

> @benpatterson <https://github.com/benpatterson> putting reset_db back in
> scripts/reset-test-db.sh makes this build green. It looks to me Jenkins
> passes PR when scripts/reset-test-db.sh is broken.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/11097#issuecomment-168179933>.
>

ziafazal commented:
@benpatterson I have tested it with both of these cases.
1. Using the serversonly/testsonly combination
2. Running as a single process

benpatterson commented:
@ziafazal Awesome. Thanks for the fix! :+1:

mattdrayer commented:
Jumping onto this PR so I remember to merge it in the morning.

benpatterson commented:
@mattdrayer I'm happy to merge. 

Heads up @raeeschachar this should resolve some various errors when doing bok-choy test development with the serversonly/testsonly peradigm.

raeeschachar commented:
Perfect!
